Add muddy points day 2- scRNAseq quantification (#36)
AlexsLemonade/2021-june-training: supplemental_notebooks/2021-06-29_muddiest_points.Rmd

library(magrittr)

# Packages for single cell processing
library(scater)
library(scran)
```

```{r}
## set up file paths 
# main data directory
data_dir <- file.path("data", "hodgkins")

# Normalized count matrix file 
normalized_sce_file <- file.path(data_dir, "normalized", "normalized_hodgkins_sce.rds")
```

```{r}
# read in normalized hodgkins sce object
hodgkins_sce <- readr::read_rds(hodgkins_sce_file)
```

We will go ahead and use the `scater::librarySizeFactors` function to compute the size factors for our counts matrix.
Then we can use the same `scater::logNormCounts` that we used in class to apply the size factors to the counts matrix, but here we will have to specify the vector containing the size factors we would like to use since they are not in the `colData` yet. 

```{r}
# calculate library size factors on counts matrix from hodgkins_sce
library_sf <- scater::librarySizeFactors(counts(hodgkins_sce))

# normalize and log transform counts using library size factors 
library_sf_sce <- scater::logNormCounts(hodgkins_sce, size.factors = library_sf)
```

Let's take a look now and compare these two different types of analysis using our PCA analysis. 

> Rather than use `calculatePCA()` though we can use `runPCA()` and add the information directly to our two `SingleCellExperiments`. 

Reviewer: Since we have now shown this a few times, we can simplify.
```suggestion
We will use `runPCA()` and add the results directly to our two `SingleCellExperiments`. 
```

```{r}
# calculate PCA for both normalized SingleCellExperiment objects
library_sf_sce <- scater::runPCA(library_sf_sce)
pool_sce <- scater::runPCA(hodgkins_sce)
```

```{r}
# plot PCA of normalized counts using library size scaling
scater::plotReducedDim(library_sf_sce, dimred = "PCA",
                       colour = "detected") + 
  ggtitle("Library Size Scaling")
```
```{r}
# plot PCA of normalized counts using pooling of size factors 
scater::plotReducedDim(pool_sce, dimred = "PCA",
                       colour = "detected") +
  ggtitle("Pooled Size Scaling/ Deconvolution")
```

> As you can see the normalized matrix with library scaling in comparison to the second plot where we have performed clustering prior to calculating size factors results in a different PCA plot. 

Reviewer: ```suggestion
As you can see, the library scaling and pooled scaling result in a quite different PCA plots with different clustering patterns, and it is reasonable to assume that these difference will also affect other downstream analyses. 
```

library_sf <- scater::librarySizeFactors(counts(hodgkins_sce))

# normalize and log transform counts using library size factors 
library_sf_sce <- scater::logNormCounts(hodgkins_sce, size.factors = library_sf)
```
Since the original `SingleCellExperiment` already has the normalized counts using the pooled scaling method class, we don't need to compute that again now. 

Let's take a look now and compare these two different types of analysis using our PCA analysis. 
We will use `runPCA()` and add the results directly to our two `SingleCellExperiments`. 

```{r}
# calculate PCA for both normalized SingleCellExperiment objects
library_sf_sce <- scater::runPCA(library_sf_sce)
pool_sce <- scater::runPCA(hodgkins_sce)
```

```{r}
# plot PCA of normalized counts using library size scaling
scater::plotReducedDim(library_sf_sce, dimred = "PCA",
                       colour = "detected") + 
  ggtitle("Library Size Scaling")
```
```{r}
# plot PCA of normalized counts using pooling of size factors 
scater::plotReducedDim(pool_sce, dimred = "PCA",
                       colour = "detected") +
  ggtitle("Pooled Size Scaling/ Deconvolution")
```

> As you can see, the library scaling and pooled scaling result in a quite different PCA plots with different clustering patterns, and it is reasonable to assume that these difference will also affect other downstream analyses. 

Reviewer: ```suggestion
As you can see, the library scaling and pooled scaling result in a quite different PCA plots with different clustering patterns, and it is reasonable to assume that these differences will also affect other downstream analyses. 
```
